FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/Services/AuthService.swift

```diff
@@ -167,21 +173,37 @@ public final class AuthService {
     }
   }
 
+  public func handleAutoUpgradeAnonymousUser(credentials credentials: AuthCredential) async throws {
+    do {
+      try await currentUser?.link(with: credentials)
```

Comment:
We should handle the `currentUser` null state in a separate conditional, otherwise this do-catch has the possibility of executing nothing and returning no error. From a convenience perspective it may also be nice to pass the unwrapped user through to the `AccoutMergeConflictContext` so the consumer doesn't have to again unwrap `currentUser`.

In theory `currentUser` should never be null here, so throwing an error is appropriate.

Reply:
@morganchen12 - updated. I didn't add User to the exception as it produced an Xcode compiler error: Stored property 'user' of 'Sendable'-conforming struct 'AccountMergeConflictContext' has non-sendable type 'User'

It seems User is non-sendable.

Reply:
A fix for that is in the pipeline in upstream Firebase, so maybe leave a TODO and just send the user's UID (String) for now?

Reply:
Done, see [`a014c7a` (#1247)](https://github.com/firebase/FirebaseUI-iOS/pull/1247/commits/a014c7a548f11031ee086877c1e37f9e43790d2d)